### ISSUE TITLE
repr of a PkgId isn't usable

### ISSUE THREAD

**Issue by IanButterworth** [JuliaLang/julia#52793]
```
julia> repr(Base.PkgId(Base.UUID("3fa0cd96-eef1-5676-8a61-b3b8758bbffb"), "REPL"))
"REPL [3fa0cd96-eef1-5676-8a61-b3b8758bbffb]"
```

**Comment by KristofferC** [comment]
Yeah, the 2 arg show should be made into a 3-arg. I encounter this all the time.

**Comment by vtjnash** [comment]
I don't know that we consider this a bug. I think the only requirement for 2-arg show (and therefore repr) is that it be simple and unambiguous, which this already is

**Comment by KristofferC** [comment]
I am not sure anyone is sayins this is a bug, but the docs on `repr` has this to say:

 > Note that repr(x) is usually similar to how the value of x would be entered in Julia. See also repr(MIME("text/plain"), x) to instead return a "pretty-printed" version of x designed more for human consumption, equivalent to the REPL display of x.

And `show`:

> The representation used by `show` generally includes Julia-specific formatting and type information, and should be parseable Julia code when possible.
